packages/calcite-components/src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -635,28 +632,39 @@ export class InputTimePicker
   }
 
   private delocalizeTimeStringToParts(
-    localizedTimeString: string,
+    value: string,
     fractionalSecondFormatToken?: "S" | "SS" | "SSS",
   ): DayjsTimeParts {
-    const ltsFormatString = this.localeConfig?.formats?.LTS;
-    const fractionalSecondTokenMatch = ltsFormatString.match(/ss\.*(S+)/g);
+    let localizedTimeString = value;
+    const effectiveHourFormat = isLocaleHourFormatOpposite(this.hourFormat, this.effectiveLocale)
+      ? getLocaleOppositeHourFormat(this.effectiveLocale)
+      : getLocaleHourFormat(this.effectiveLocale);
 
-    if (fractionalSecondFormatToken && this.shouldIncludeFractionalSeconds()) {
-      const secondFormatToken = `ss.${fractionalSecondFormatToken}`;
-      this.localeConfig.formats.LTS = fractionalSecondTokenMatch
-        ? ltsFormatString.replace(fractionalSecondTokenMatch[0], secondFormatToken)
-        : ltsFormatString.replace("ss", secondFormatToken);
-    } else if (fractionalSecondTokenMatch) {
-      this.localeConfig.formats.LTS = ltsFormatString.replace(fractionalSecondTokenMatch[0], "ss");
+    if (
+      localizedTwentyFourHourMeridiems.has(this.effectiveLocale) &&
+      effectiveHourFormat === "12"
+    ) {
+      const localizedAM = localizedTwentyFourHourMeridiems.get(this.effectiveLocale).am;
+      const localizedPM = localizedTwentyFourHourMeridiems.get(this.effectiveLocale).pm;
+      const localizedAMRegEx = new RegExp(String.raw`${localizedAM}`, "g");
```

Comment:
Looking at the `am`, `pm` values, would a string lookup work as well?

Reply:
It looks like `String.prototype.includes()` also works.  Is that what you meant by string lookup?